test/integration/context-based-restrictions.v1.test.js

```diff
@@ -0,0 +1,934 @@
+      transactionId: uuidv4(),
+    };
+
+    await expect(contextBasedRestrictionsService.getZone(params)).rejects.toMatchObject({
+      status: 404,
+    });
+  });
+
+  test('replaceZone() - 200', async () => {
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'ipAddress',
+      value: '169.24.57.235',
+    };
+
+    const params = {
+      zoneId,
+      ifMatch: zoneEtag,
+      name: 'an example of updated zone',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of updated zone',
+      transactionId: uuidv4(),
+    };
+
+    const res = await contextBasedRestrictionsService.replaceZone(params);
+    expect(res).toBeDefined();
+    expect(res.status).toBe(200);
+    expect(res.result).toBeDefined();
+  });
+
+  test('replaceZone() - 400 - invalid type: Address', async () => {
+    // The zone could not be updated due to invalid type 'Address'.
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'Address',
+      value: '169.24.57.235',
+    };
+
+    const params = {
+      zoneId,
+      ifMatch: zoneEtag,
+      name: 'an example of an invalid zone',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of an invalid zone',
+      transactionId: uuidv4(),
+    };
+
+    await expect(contextBasedRestrictionsService.replaceZone(params)).rejects.toMatchObject({
+      status: 400,
+    });
+  });
+
+  test('replaceZone() - 400 - missing required parameter: ifMatch', async () => {
+    // The zone could not be updated due to missing required parameter: ifMatch'.
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'Address',
+      value: '169.24.57.235',
+    };
+
+    const params = {
+      zoneId,
+      name: 'an example of an invalid zone',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of an invalid zone',
+      transactionId: uuidv4(),
+    };
+
+    await expect(contextBasedRestrictionsService.replaceZone(params)).rejects.toMatchObject({
+      'message': 'Missing required parameters: ifMatch',
+    });
+  });
+
+  test('replaceZone() - 400 - zone not found', async () => {
+    // The zone could not be updated due to zone is not found.
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'Address',
+      value: '169.24.57.235',
+    };
+
+    const params = {
+      zoneId: invalidID,
+      ifMatch: zoneEtag,
+      name: 'an example of zone not found',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of zone not found',
+      transactionId: uuidv4(),
+    };
+
+    await expect(contextBasedRestrictionsService.replaceZone(params)).rejects.toMatchObject({
+      status: 400,
+    });
+  });
+
```

Comment:
The Go SDK contains a test against `412`. I'd like to ask to add a test here covering `412`.

Reply:
add new test case 'replaceZone() - 412 - invalid IfMath parameter"